DSharpPlus.CommandsNext/CommandsNextExtension.cs

```diff
@@ -489,8 +489,19 @@ private void RegisterCommands(Type t, CommandGroupBuilder? currentParent, IEnume
                         foreach (var chk in inheritedChecks)
                             groupBuilder.WithExecutionCheck(chk);
 
-                        foreach (var mi in ti.DeclaredMethods.Where(x => x.IsCommandCandidate(out _) && x.GetCustomAttribute<GroupCommandAttribute>() != null))
-                            groupBuilder.WithOverload(new CommandOverloadBuilder(mi));
+                        if (ti.DeclaredMethods.Any(x => x.IsCommandCandidate(out _) && x.GetCustomAttribute<GroupCommandAttribute>() != null))
+                        {
+                            var groupCommandOverloads =
+                                ti.DeclaredMethods.Where(x => x.IsCommandCandidate(out _) && x.GetCustomAttribute<GroupCommandAttribute>() != null);
```

Comment:
Feels a bit daft to not merge these into one, or just use `Count(Func<T, bool>)`

Reply:
this is solely a more-or-less pointless optimization; to never invoke any of that logic if no group commands are present in the first place

Reply:
Yes, but I don't see how this invalidates my point considering that in experience `.Count` is faster then `.Where().Count()`, which is being done below 😄

Reply:
Where's also the fact that you can slightly save on allocations by not allocating two iterators.

Reply:
i suppose so, though invoking a `.Count()` just for it to be zero in a lot of cases, likely most cases doesn't seem right either.

Reply:
You're forcing an enumeration in all cases, so trying to guard with `.Any()` isn't a possible solution. I think we should just bite the bullet and check count anyway